cpp/Platform.Timestamps/Timestamp.h

```diff
@@ -1,25 +1,73 @@
-﻿namespace Platform::Timestamps
-{
-    struct Timestamp : public IEquatable<Timestamp>
-    {
-        public: inline static std::string DefaultFormat = "yyyy.MM.dd hh:mm:ss.fffffff";
-
-        public: std::uint64_t Ticks = 0;
+﻿#pragma once
 
-        public: Timestamp(std::uint64_t ticks) { Ticks = ticks; }
+#include "CommonEraClock.h"
+#include <functional>
+#include <iomanip>
+#include <sstream>
+#include <string>
 
-        public: Timestamp(DateTime dateTime) : Timestamp((std::uint64_t)dateTime.ToUniversalTime().Ticks) { }
-
-        public: operator DateTime() const { return DateTime((std::int64_t)this->Ticks, DateTimeKind.Utc); }
+namespace Platform::Timestamps
+{
+    class Timestamp
+    {
+    public:
+        static constexpr const char* DefaultFormat = "%Y.%m.%d %H:%M:%S";
+        static constexpr int TicksPerSecond = 10000000;
 
-        public: Timestamp(std::uint64_t ticks) : Timestamp(ticks) { }
+        Timestamp(const uint64_t& ticks)
+            : _ticks(ticks)
+        {
+        }
+        Timestamp(const std::chrono::time_point<CommonEraClock>& common_time_point)
+            : _ticks(common_time_point.time_since_epoch().count())
+        {
+        }
+        Timestamp(const Timestamp& timestamp)
+            : _ticks(timestamp.Ticks())
+        {
+        }
+        Timestamp()
+            : _ticks(0)
+        {
+        }
 
-        public: operator ulong() const { return this->Ticks; }
+        uint64_t Ticks() const noexcept
+        {
+            return _ticks;
+        }
+        void Ticks(const uint64_t& ticks)
+        {
+            _ticks = ticks;
+        }
 
-        public: override std::string ToString() { return ((DateTime)this).ToString(DefaultFormat); }
+        bool operator==(const Timestamp& other) const
```

Comment:
dirty c++20 magic
```cpp
bool operator==(const Timestamp& other) const noexcept = default;
```
and, also, can be constexpr'ed. Looks like 99% methods can be constexpred